### PR TITLE
Add list multipart uploads

### PR DESCRIPTION
**Pull request by Tomju2**
Added ListMultipartUploads 

Includes ListMultipartUploads and Upload structs.

### CODE REVIEW

File: include/baseclient.h [hunk 1]
```diff
@@ -111,6 +111,8 @@ class BaseClient {
       IsObjectLegalHoldEnabledArgs args);
   ListBucketsResponse ListBuckets(ListBucketsArgs args);
   ListBucketsResponse ListBuckets();
+  ListMultipartUploadsResponse  ListMultipartUploads(ListMultipartUploadArgs args);
+  ListMultipartUploadsResponse  ListMultipartUploads();
```
Comment (balamurugana) [inline include/baseclient.h:115]: This method is not needed because we must specify bucket name.

File: include/args.h [hunk 2]
```diff
@@ -104,6 +104,17 @@ struct CreateMultipartUploadArgs : public ObjectArgs {
...
 
+struct ListMultipartUploadArgs : public BucketArgs{
+  std::string delimiter;
+  std::string encoding_type;
+  std::string key_marker;
+  unsigned int max_uploads = 1000;
+  std::string prefix;
+  std::string upload_id_marker;
+  utils::Multimap extra_headers;
+  utils::Multimap extra_query_params;
```
Comment (balamurugana) [inline include/args.h:115]: `extra_headers` and `extra_query_params` are duplicates, already there in `BaseArgs`

File: include/args.h [hunk 3]
```diff
@@ -104,6 +104,17 @@ struct CreateMultipartUploadArgs : public ObjectArgs {
   utils::Multimap headers;
 };  // struct CreateMultipartUploadArgs
 
+struct ListMultipartUploadArgs : public BucketArgs{
```
Comment (balamurugana) [inline include/args.h:107]: Practically we need [ListObjectsCommonArgs](https://github.com/minio/minio-cpp/blob/main/include/args.h#L178) to be extended, but overriding `max_keys` as `max_uploads`, then new fields `key-marker`/`upload-id-marker` need to be added

File: include/response.h [hunk 4]
```diff
@@ -202,6 +202,48 @@ struct Item : public Response {
...
+  std::string encoding_type;
+  std::string object_name;
+  std::string upload_id;
+  std::string initiator_id;
+  std::string initiator_name;
+  std::string tag;
+  std::string owner_id;
+  std::string owner_name;
+  std::string storage_class;
+  std::string initiated_time;
```
Comment (balamurugana) [inline include/response.h:215]: Below are the fields
```
std::string checksum_algorithm;
utils::Time initiated;
std::string initiator_display_name;
std::string initiator_id;
std::string key;
std::string owner_display_name;
std::string owner_id;
std::string storage_class;
std::string upload_id;

std::string prefix;
```

File: include/response.h [hunk 5]
```diff
@@ -202,6 +202,48 @@ struct Item : public Response {
...
+struct ListMultipartUploadsResponse : public Response{
+  // Common
+  std::string bucket_name;
+  std::string key_marker;
+  std::string next_upload_id_marker;
+  std::string next_key_marker;
+  std::string upload_id_marker;
+  unsigned int max_uploads;
+  bool is_truncated;
+  std::string encoding_type;
```
Comment (balamurugana) [inline include/response.h:233]: These fields are missing
```
std::string prefix;
std::string delimiter;
```

File: src/baseclient.cc [hunk 6]
```diff
@@ -1021,6 +1021,45 @@ minio::s3::ListBucketsResponse minio::s3::BaseClient::ListBuckets() {
...
+  Request req(http::Method::kGet, region, base_url_,
+              args.extra_headers, args.extra_query_params);
+  
+  req.bucket_name = args.bucket;
+
+  req.query_params.Add("uploads","");
+  req.query_params.Add("delimiter","");
+  req.query_params.Add("max-uploads", "1000");
+  req.query_params.Add("prefix",args.prefix);
+  req.query_params.Add("encoding-type","url");
```
Comment (balamurugana) [inline src/baseclient.cc:1043]: You would need to use `delimiter`/`max-uploads`/`prefix`/`encoding-type` parameters from `args`

File: include/args.h (hunk 3 above)
Comment (Tomju2) [reply include/args.h:107]: Ok, so should I delete ListMultipartUploadArgs and pass that arguments to ListObjectsCommonArgs right?

**Pull request closed without merging**